workerapi/server/server.go

```diff
@@ -0,0 +1,102 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/luci/go-render/render"
+	"github.com/scootdev/scoot/workerapi/gen-go/worker"
+)
+
+func Serve(handler worker.Worker, addr string, transportFactory thrift.TTransportFactory, protocolFactory thrift.TProtocolFactory) error {
+	transport, err := thrift.NewTServerSocket(addr)
+	if err != nil {
+		return err
+	}
+	processor := worker.NewWorkerProcessor(handler)
+	server := thrift.NewTSimpleServer4(processor, transport, transportFactory, protocolFactory)
+
+	fmt.Println("About to serve")
+
+	return server.Serve()
+}
+
+type Handler struct {
+	runs map[string]*worker.RunStatus
+}
+
+func NewHandler() worker.Worker {
+	return &Handler{make(map[string]*worker.RunStatus)}
+}
+
+func (h *Handler) QueryWorker() (*worker.WorkerStatus, error) {
+	wstatus := worker.NewWorkerStatus()
+	wstatus.Running = []string{}
+	for runId, status := range h.runs {
+		if status.Status == worker.Status_RUNNING {
+			wstatus.Running = append(wstatus.Running, runId)
+		}
+	}
+	return wstatus, nil
+}
+
+func makeStatus(status worker.Status, exitCode int32, info string) *worker.RunStatus {
+	rs := worker.NewRunStatus()
+	rs.Status = status
+	rs.Info = &info
+	rs.ExitCode = &exitCode
+	return rs
+}
+
+//TODO: integrate runner lib and remove all this temp test code.
```

Comment:
Oh, I see. Yeah, I'd think it might be easier to just integrate it now, but I don't want to block you.